diff --git a/neural/neuralbyteprediction.pas b/neural/neuralbyteprediction.pas
index 6da031f..55ab9b6 100644
--- a/neural/neuralbyteprediction.pas
+++ b/neural/neuralbyteprediction.pas
@@ -485,7 +485,7 @@ begin
   if (NG.TestNeuronLayer.N > 10) then
   begin
     NG.TestNeuronLayer.TestThreshold :=
-    NG.TestNeuronLayer.N - Random(NG.TestNeuronLayer.N div 10);
+    NG.TestNeuronLayer.N - Random(LongInt(NG.TestNeuronLayer.N div 10));
   end
   else
   begin
